android/modules/network/src/java/ti/modules/titanium/network/TiHTTPClient.java

```diff
@@ -1171,7 +1170,7 @@ public void send(Object userData) throws MethodNotSupportedException
 				boolean queryStringAltered = false;
 				for (String key : data.keySet()) {
 					Object value = data.get(key);
-					if (isPostOrPut && (value != null)) {
+					if (isPostOrPutOrPatch && (value != null)) {
```

Comment:
I personally hate this, technically any request is permitted a request body with the HTTP spec. I've seen API's that support `DELETE` with request body parameters.

I suppose this is not your beef though, and should be addressed separately.

Looks good :+1: